### PR TITLE
How-to store CAR files

### PR DESCRIPTION
**Pull request by yusefnapora**
This is a start on the how-to portion of #146.

Leaving as draft until:

- [x] links and example for js-car 
- [x] links ~~and example~~ for go-car

### CODE REVIEW

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-tos/store.md`:
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...s such as fetch  (opens new window) and File  (opens new window). This guide...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ....storage package to your dependencies:  First we need to create a Web3.Stora...
                                             ^^^^^
  ```
- The phrase ‘a lot of’ might be wordy. Consider using “many”.
  ```
  ...ent method: If you expect to be loading a lot of large files, you may be bet...
                                             ^^^^^^^^
  ```
- Apostrophe might be missing. Did you mean “file’s” (curly possessive apostrophe) or “file's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ...one content archive, and linking to the files inside is much easier if each ...
                                             ^^^^^
  ```
- Possibly missing comma found.
  ```
  ...an array of Files, uploading is simple: IMPORTANT Deleting files from the We...
                                             ^^^^^^^^^
  ```
- Possibly missing article found.
  ```
  ... IMPORTANT Deleting files from the Web3.Storage site's Files page  (opens ne...
                                             ^^^^^^^
  ```
- It seems that a comma is missing.
  ```
  ... progress of an upload to the console:  So far we've focused on using the pu...
                                             ^^^^^^
  ```
- Use ‘own’ only if you want to emphasize that something belongs to a person or thing or was done by a person or thing. Consider writing “your”.
  ```
  ... structure your data, you can construct your own CAR files and upload them d...
                                             ^^^^^^^^
  ```
- Consider using a more formal alternative to avoid the overused word “more”.
  ```
  ...s. The Working with CAR files guide has more information about the @ipld/car...
                                             ^^^^^^^^^^^^^^^^
  ```
- Consider using a more formal alternative to avoid the overused word “more”.
  ```
  ...to guide on retrieval. You can also get more information about the status of...
                                             ^^^^^^^^^^^^^^^^
  ```
### `docs/how-tos/work-with-car-files.md`:
- The following link failed to connect:
  https://nodejs.com
- The term ‘Storage client’ seems to be a compound noun. Make sure that an article or a determiner is not missing before the compound noun.
  ```
  ...ow for more information.  When the Web3.Storage client packs up regular file...
                                             ^^^^^^^^^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...64-encoded when using dag-json. Gateway support Although Web3.Storage suppor...
                                             ^^^^^^^
  ```
- As a shorter alternative for ‘able to’, consider using “can not”.
  ```
  ...not currently "speak" these formats and is not able to provide such data ove...
                                             ^^^^^^^^^^^^^^
  ```

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-tos/store.md`:
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...s such as fetch  (opens new window) and File  (opens new window). This guide...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ....storage package to your dependencies:  First we need to create a Web3.Stora...
                                             ^^^^^
  ```
- The phrase ‘a lot of’ might be wordy. Consider using “many”.
  ```
  ...ent method: If you expect to be loading a lot of large files, you may be bet...
                                             ^^^^^^^^
  ```
- Apostrophe might be missing. Did you mean “file’s” (curly possessive apostrophe) or “file's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ...one content archive, and linking to the files inside is much easier if each ...
                                             ^^^^^
  ```
- Possibly missing comma found.
  ```
  ...an array of Files, uploading is simple: IMPORTANT Deleting files from the We...
                                             ^^^^^^^^^
  ```
- Possibly missing article found.
  ```
  ... IMPORTANT Deleting files from the Web3.Storage site's Files page  (opens ne...
                                             ^^^^^^^
  ```
- It seems that a comma is missing.
  ```
  ... progress of an upload to the console:  So far we've focused on using the pu...
                                             ^^^^^^
  ```
- Use ‘own’ only if you want to emphasize that something belongs to a person or thing or was done by a person or thing. Consider writing “your”.
  ```
  ... structure your data, you can construct your own CAR files and upload them d...
                                             ^^^^^^^^
  ```
- Consider using a more formal alternative to avoid the overused word “more”.
  ```
  ...s. The Working with CAR files guide has more information about the @ipld/car...
                                             ^^^^^^^^^^^^^^^^
  ```
- Consider using a more formal alternative to avoid the overused word “more”.
  ```
  ...to guide on retrieval. You can also get more information about the status of...
                                             ^^^^^^^^^^^^^^^^
  ```
### `docs/how-tos/work-with-car-files.md`:
- The following link failed to connect:
  https://nodejs.com
- The term ‘Storage client’ seems to be a compound noun. Make sure that an article or a determiner is not missing before the compound noun.
  ```
  ...ow for more information.  When the Web3.Storage client packs up regular file...
                                             ^^^^^^^^^^^^^^
  ```
- Possibly missing comma found.
  ```
  ... an example that uses dag-json. Gateway support Although Web3.Storage suppor...
                                             ^^^^^^^
  ```
- As a shorter alternative for ‘able to’, consider using “can not”.
  ```
  ...not currently "speak" these formats and is not able to provide such data ove...
                                             ^^^^^^^^^^^^^^
  ```
### `docs/reference/client-library.md`:
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain a fre...
                                             ^^^^^^^
  ```
- ‘put’ seems less likely than ‘out’.
  ```
  ... can be used as parameters when calling put(): name String. The name paramet...
                                             ^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ... as parameters when calling put(): name String. The name parameter lets you ...
                                             ^^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...r IPFS path. For example, if you have a files directory like this: Using the...
                                             ^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...turn a CID for a directory containing a files subdirectory, like this: Howev...
                                             ^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...ou do this instead: The contents of the files directory will be at the top l...
                                             ^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...ill be at the top level, instead of the files directory itself:  onRootCidRe...
                                             ^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...ow) containing the queried CID. created String. The created property gives t...
                                             ^^^^^^
  ```
- Apostrophe might be missing. Did you mean “pin’s” (curly possessive apostrophe) or “pin's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ...pload the same file(s). pins Array. The pins property is an array of Pin obj...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...operty is an array of Pin objects. Each Pin object represents a specific pee...
                                             ^^^
  ```
- Apostrophe might be missing. Did you mean “deal’s” (curly possessive apostrophe) or “deal's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ... following structure:  deals Array. The deals property is an array of Deal o...
                                             ^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...perty is an array of Deal objects. Each Deal object represents a specific st...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...t with the following properties: before String. Specifies a date, in ISO 860...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 car The CAR file to be uploaded. 2 {...
                                             ^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ... parameters when calling putCar(): name String. The name parameter lets you ...
                                             ^^^^^^
  ```
- Insert a space between the numerical value and the unit symbol.
  ```
  ...lt, data is split into chunks of around 10MB.  decoders BlockDecoder  (opens...
                                             ^^^^
  ```

**Comment by yusefnapora** [comment]
I think this is ready to review. I wasn't able to find a good example of using `go-car` lying around, so if anyone knows where to find one, let me know :)

Also, let me know if I'm getting too into the weeds with IPLD stuff (or not weedy enough 😄 ). I tried to keep the explanation at a fairly high level, but there's a few new technical terms sprinkled around that might be confusing (codec, etc.)

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-tos/store.md`:
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...s such as fetch  (opens new window) and File  (opens new window). This guide...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ....storage package to your dependencies:  First we need to create a Web3.Stora...
                                             ^^^^^
  ```
- The phrase ‘a lot of’ might be wordy. Consider using “many”.
  ```
  ...ent method: If you expect to be loading a lot of large files, you may be bet...
                                             ^^^^^^^^
  ```
- Apostrophe might be missing. Did you mean “file’s” (curly possessive apostrophe) or “file's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ...one content archive, and linking to the files inside is much easier if each ...
                                             ^^^^^
  ```
- Possibly missing comma found.
  ```
  ...an array of Files, uploading is simple: IMPORTANT Deleting files from the We...
                                             ^^^^^^^^^
  ```
- Possibly missing article found.
  ```
  ... IMPORTANT Deleting files from the Web3.Storage site's Files page  (opens ne...
                                             ^^^^^^^
  ```
- It seems that a comma is missing.
  ```
  ... progress of an upload to the console:  So far we've focused on using the pu...
                                             ^^^^^^
  ```
- Use ‘own’ only if you want to emphasize that something belongs to a person or thing or was done by a person or thing. Consider writing “your”.
  ```
  ... structure your data, you can construct your own CAR files and upload them d...
                                             ^^^^^^^^
  ```
- Consider using a more formal alternative to avoid the overused word “more”.
  ```
  ...s. The Working with CAR files guide has more information about the @ipld/car...
                                             ^^^^^^^^^^^^^^^^
  ```
- Consider using a more formal alternative to avoid the overused word “more”.
  ```
  ...to guide on retrieval. You can also get more information about the status of...
                                             ^^^^^^^^^^^^^^^^
  ```
### `docs/how-tos/work-with-car-files.md`:
- The following link failed to connect:
  https://nodejs.com
- The term ‘Storage client’ seems to be a compound noun. Make sure that an article or a determiner is not missing before the compound noun.
  ```
  ...ow for more information.  When the Web3.Storage client packs up regular file...
                                             ^^^^^^^^^^^^^^
  ```
- Possibly missing comma found.
  ```
  ... an example that uses dag-json. Gateway support Although Web3.Storage suppor...
                                             ^^^^^^^
  ```
- As a shorter alternative for ‘able to’, consider using “can not”.
  ```
  ...not currently "speak" these formats and is not able to provide such data ove...
                                             ^^^^^^^^^^^^^^
  ```
### `docs/reference/client-library.md`:
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain a fre...
                                             ^^^^^^^
  ```
- ‘put’ seems less likely than ‘out’.
  ```
  ... can be used as parameters when calling put(): name String. The name paramet...
                                             ^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ... as parameters when calling put(): name String. The name parameter lets you ...
                                             ^^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...r IPFS path. For example, if you have a files directory like this: Using the...
                                             ^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...turn a CID for a directory containing a files subdirectory, like this: Howev...
                                             ^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...ou do this instead: The contents of the files directory will be at the top l...
                                             ^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...ill be at the top level, instead of the files directory itself:  onRootCidRe...
                                             ^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...ow) containing the queried CID. created String. The created property gives t...
                                             ^^^^^^
  ```
- Apostrophe might be missing. Did you mean “pin’s” (curly possessive apostrophe) or “pin's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ...pload the same file(s). pins Array. The pins property is an array of Pin obj...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...operty is an array of Pin objects. Each Pin object represents a specific pee...
                                             ^^^
  ```
- Apostrophe might be missing. Did you mean “deal’s” (curly possessive apostrophe) or “deal's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ... following structure:  deals Array. The deals property is an array of Deal o...
                                             ^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...perty is an array of Deal objects. Each Deal object represents a specific st...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...t with the following properties: before String. Specifies a date, in ISO 860...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 car The CAR file to be uploaded. 2 {...
                                             ^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ... parameters when calling putCar(): name String. The name parameter lets you ...
                                             ^^^^^^
  ```
- Insert a space between the numerical value and the unit symbol.
  ```
  ...lt, data is split into chunks of around 10MB.  decoders BlockDecoder  (opens...
                                             ^^^^
  ```

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-tos/store.md`:
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...s such as fetch  (opens new window) and File  (opens new window). This guide...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ....storage package to your dependencies:  First we need to create a Web3.Stora...
                                             ^^^^^
  ```
- The phrase ‘a lot of’ might be wordy. Consider using “many”.
  ```
  ...ent method: If you expect to be loading a lot of large files, you may be bet...
                                             ^^^^^^^^
  ```
- Apostrophe might be missing. Did you mean “file’s” (curly possessive apostrophe) or “file's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ...one content archive, and linking to the files inside is much easier if each ...
                                             ^^^^^
  ```
- Possibly missing comma found.
  ```
  ...an array of Files, uploading is simple: IMPORTANT Deleting files from the We...
                                             ^^^^^^^^^
  ```
- Possibly missing article found.
  ```
  ... IMPORTANT Deleting files from the Web3.Storage site's Files page  (opens ne...
                                             ^^^^^^^
  ```
- It seems that a comma is missing.
  ```
  ... progress of an upload to the console:  So far we've focused on using the pu...
                                             ^^^^^^
  ```
- Use ‘own’ only if you want to emphasize that something belongs to a person or thing or was done by a person or thing. Consider writing “your”.
  ```
  ... structure your data, you can construct your own CAR files and upload them d...
                                             ^^^^^^^^
  ```
- Consider using a more formal alternative to avoid the overused word “more”.
  ```
  ...s. The Working with CAR files guide has more information about the @ipld/car...
                                             ^^^^^^^^^^^^^^^^
  ```
- Consider using a more formal alternative to avoid the overused word “more”.
  ```
  ...to guide on retrieval. You can also get more information about the status of...
                                             ^^^^^^^^^^^^^^^^
  ```
### `docs/how-tos/work-with-car-files.md`:
- The following link returned HTTP status code 503 (Service Unavailable):
  https://web3.storage
- The following link returned HTTP status code 503 (Service Unavailable):
  https://web3.storage/about/
- The following link failed to connect:
  https://nodejs.com
- The term ‘Storage client’ seems to be a compound noun. Make sure that an article or a determiner is not missing before the compound noun.
  ```
  ...ow for more information.  When the Web3.Storage client packs up regular file...
                                             ^^^^^^^^^^^^^^
  ```
- Possibly missing comma found.
  ```
  ... an example that uses dag-json. Gateway support Although Web3.Storage suppor...
                                             ^^^^^^^
  ```
- As a shorter alternative for ‘able to’, consider using “can not”.
  ```
  ...not currently "speak" these formats and is not able to provide such data ove...
                                             ^^^^^^^^^^^^^^
  ```
### `docs/reference/client-library.md`:
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain a fre...
                                             ^^^^^^^
  ```
- ‘put’ seems less likely than ‘out’.
  ```
  ... can be used as parameters when calling put(): name String. The name paramet...
                                             ^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ... as parameters when calling put(): name String. The name parameter lets you ...
                                             ^^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...r IPFS path. For example, if you have a files directory like this: Using the...
                                             ^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...turn a CID for a directory containing a files subdirectory, like this: Howev...
                                             ^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...ou do this instead: The contents of the files directory will be at the top l...
                                             ^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...ill be at the top level, instead of the files directory itself:  onRootCidRe...
                                             ^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...ow) containing the queried CID. created String. The created property gives t...
                                             ^^^^^^
  ```
- Apostrophe might be missing. Did you mean “pin’s” (curly possessive apostrophe) or “pin's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ...pload the same file(s). pins Array. The pins property is an array of Pin obj...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...operty is an array of Pin objects. Each Pin object represents a specific pee...
                                             ^^^
  ```
- Apostrophe might be missing. Did you mean “deal’s” (curly possessive apostrophe) or “deal's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ... following structure:  deals Array. The deals property is an array of Deal o...
                                             ^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...perty is an array of Deal objects. Each Deal object represents a specific st...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...t with the following properties: before String. Specifies a date, in ISO 860...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 car The CAR file to be uploaded. 2 {...
                                             ^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ... parameters when calling putCar(): name String. The name parameter lets you ...
                                             ^^^^^^
  ```
- Insert a space between the numerical value and the unit symbol.
  ```
  ...lt, data is split into chunks of around 10MB.  decoders BlockDecoder  (opens...
                                             ^^^^
  ```

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/how-tos/store.md`:
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...s such as fetch  (opens new window) and File  (opens new window). This guide...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ....storage package to your dependencies:  First we need to create a Web3.Stora...
                                             ^^^^^
  ```
- The phrase ‘a lot of’ might be wordy. Consider using “many”.
  ```
  ...ent method: If you expect to be loading a lot of large files, you may be bet...
                                             ^^^^^^^^
  ```
- Apostrophe might be missing. Did you mean “file’s” (curly possessive apostrophe) or “file's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ...one content archive, and linking to the files inside is much easier if each ...
                                             ^^^^^
  ```
- Possibly missing comma found.
  ```
  ...an array of Files, uploading is simple: IMPORTANT Deleting files from the We...
                                             ^^^^^^^^^
  ```
- Possibly missing article found.
  ```
  ... IMPORTANT Deleting files from the Web3.Storage site's Files page  (opens ne...
                                             ^^^^^^^
  ```
- It seems that a comma is missing.
  ```
  ... progress of an upload to the console:  So far we've focused on using the pu...
                                             ^^^^^^
  ```
- Use ‘own’ only if you want to emphasize that something belongs to a person or thing or was done by a person or thing. Consider writing “your”.
  ```
  ... structure your data, you can construct your own CAR files and upload them d...
                                             ^^^^^^^^
  ```
- Consider using a more formal alternative to avoid the overused word “more”.
  ```
  ...s. The Working with CAR files guide has more information about the @ipld/car...
                                             ^^^^^^^^^^^^^^^^
  ```
- Consider using a more formal alternative to avoid the overused word “more”.
  ```
  ...to guide on retrieval. You can also get more information about the status of...
                                             ^^^^^^^^^^^^^^^^
  ```
### `docs/how-tos/work-with-car-files.md`:
- The following link failed to connect:
  https://nodejs.com
- The term ‘Storage client’ seems to be a compound noun. Make sure that an article or a determiner is not missing before the compound noun.
  ```
  ...ow for more information.  When the Web3.Storage client packs up regular file...
                                             ^^^^^^^^^^^^^^
  ```
- Possibly missing comma found.
  ```
  ... an example that uses dag-json. Gateway support Although Web3.Storage suppor...
                                             ^^^^^^^
  ```
- As a shorter alternative for ‘able to’, consider using “can not”.
  ```
  ...not currently "speak" these formats and is not able to provide such data ove...
                                             ^^^^^^^^^^^^^^
  ```
### `docs/reference/client-library.md`:
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain a fre...
                                             ^^^^^^^
  ```
- ‘put’ seems less likely than ‘out’.
  ```
  ... can be used as parameters when calling put(): name String. The name paramet...
                                             ^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ... as parameters when calling put(): name String. The name parameter lets you ...
                                             ^^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...r IPFS path. For example, if you have a files directory like this: Using the...
                                             ^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...turn a CID for a directory containing a files subdirectory, like this: Howev...
                                             ^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...ou do this instead: The contents of the files directory will be at the top l...
                                             ^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...ill be at the top level, instead of the files directory itself:  onRootCidRe...
                                             ^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...ow) containing the queried CID. created String. The created property gives t...
                                             ^^^^^^
  ```
- Apostrophe might be missing. Did you mean “pin’s” (curly possessive apostrophe) or “pin's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ...pload the same file(s). pins Array. The pins property is an array of Pin obj...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...operty is an array of Pin objects. Each Pin object represents a specific pee...
                                             ^^^
  ```
- Apostrophe might be missing. Did you mean “deal’s” (curly possessive apostrophe) or “deal's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ... following structure:  deals Array. The deals property is an array of Deal o...
                                             ^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...perty is an array of Deal objects. Each Deal object represents a specific st...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...t with the following properties: before String. Specifies a date, in ISO 860...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 car The CAR file to be uploaded. 2 {...
                                             ^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ... parameters when calling putCar(): name String. The name parameter lets you ...
                                             ^^^^^^
  ```
- Insert a space between the numerical value and the unit symbol.
  ```
  ...lt, data is split into chunks of around 10MB.  decoders BlockDecoder  (opens...
                                             ^^^^
  ```

**Comment by yusefnapora** [comment]
Could use some 👀 on the dag-cbor example in the "working with car files" guide - it seems to work but I haven't messed with this much before and might be doing it wrong.

I also cleaned up the dag-json example that was already in the reference docs, and it runs, but the service is returning a 500:

```
node dag-json.mjs
uploading car.
file:///Users/yusef/work/projects/web3storage/web3-storage-docs/code-snippets/how-to/node_modules/web3.storage/src/lib.js:174
            throw new Error(res.message)
                  ^

Error: 500: Internal Server Error
    at pRetry.retries (file:///Users/yusef/work/projects/web3storage/web3-storage-docs/code-snippets/how-to/node_modules/web3.storage/src/lib.js:174:19)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async RetryOperation._fn (/Users/yusef/work/projects/web3storage/web3-storage-docs/code-snippets/how-to/node_modules/p-retry/index.js:50:12) {
  attemptNumber: 6,
  retriesLeft: 0
}
```

If you want to try, go to `code-snippets/how-tos` and rename `dag-json.js` to `dag-json.mjs` and run `node dag-json.mjs` - the reason for the rename is that Vuepress won't syntax highlight .mjs files when you include them as snippets, but node won't run modules correctly with a .js extension

Anyway, I can't tell if I'm doing something wrong or if the service is choking on unexpected dag-json input. 

@alanshaw / @olizilla / @vasco-santos any thoughts?

File: code-snippets/how-to/dag-cbor.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  // upload to Web3.Storage using putCar
+  const client = new Web3Storage({ token: process.env.WEB3STORAGE_TOKEN })
+  const cid = await client.putCar(reader)
+  console.log('stored dag-cbor data! CID:', cid)
+}
+
+storeDagCBOR({
+  hello: 'world'
+})
```
Comment (alanshaw) [inline code-snippets/how-to/dag-cbor.js:29]: Might be cool to return the CID and use it in another call to `storeDagCBOR` to link some data together?

Or maybe store a file, then link to it from a `dag-cbor` node `{ 'hello.txt`: CID }` and then fetch it from the gateway at `ipfs.io/ipfs/CID/hello.txt`.

File: docs/how-tos/work-with-car-files.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+The `CarReader` type shown above will read the entire contents of the CAR into memory, which may cause issues with large files. On Node.js, you can use [`CarIndexedReader`](https://github.com/ipld/js-car#carindexedreader), which reads CAR data from disk directly and uses less memory than `CarReader`.
+
+### Go
+
+The [`go-car` module](https://github.com/ipld/go-car) provides the main Golang implementation of the CAR specification. We recommend using the `v2` module version, which supports the latest version of the CAR spec.
+
+See the [API reference documentation](https://pkg.go.dev/github.com/ipld/go-car/v2) for more information.
+
+<!-- TODO: find / write simple go-car example -->
```
Comment (alanshaw) [inline docs/how-tos/work-with-car-files.md:145]: Note sure if this helps:

For writing CARs https://github.com/web3-storage/go-w3s-client/blob/main/client.go#L377

Essentially you call `car.WriteCar`, and pass it a "dag service" (key/value for CIDs that contains all the data) a root CID and a `io.Writer` (something to write the CAR bytes to). It traverses the DAG starting from the root you gave it by reading from the dag service and writes the output to the `io.Writer`.

I'm happy to sync up and explain some of this.

**Review by alanshaw — APPROVED** [review]
So cool 🚀. Lets ship!

**Comment by alanshaw** [comment]
> service is choking on unexpected dag-json input.

It shouldn't - can you open an issue 🙏

**Comment by yusefnapora** [comment]
Sweet, thanks @alanshaw :) I like your comment about linking between dag-cbor and a file - I think I'll make a new PR for that though.

**Pull request merged**